src/Service/Sqs/SqsConsumer.php

```diff
@@ -36,13 +38,15 @@ public function __construct(
         BusDriver $busDriver,
         MessageBusInterface $bus,
         SerializerInterface $serializer,
-        string $transportName,
+        SqsTransportNameResolver $transportNameResolver,
```

Comment:
I think you cannot change the order of the current arguments, this would be also a BC break.

The only solution would be to append the `$transportNameResolver` at the end (and making it nullable obviously).

And then throw an exception if both `$transportName` and `$transportNameResolver` are null ?

Reply:
Good idea, I changed it like you suggested :) Let me know if you agree with it or if there is something else, what I can improve.